### PR TITLE
Added documentation for using checkboxes to toggle collapse

### LINKED ISSUE #39728: Docs: No info about the collapse plugin supporting checkboxes as toggles
https://getbootstrap.com/docs/5.3/components/collapse/ only tasks about buttons and `<a>` elements as toggles. However, using checkboxes also works fine, like in the following code provided by [ChatGPT](https://chat.openai.com). Therefore, it should be pointed out in the docs that you can also use checkboxes as collapse toggles.
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css">
    <title>Checkbox Beispiel</title>
</head>
<body>

<div class="container mt-5">
    <div class="form-check">
        <input class="form-check-input" type="checkbox" id="hideCheckbox" data-bs-toggle="collapse" data-bs-target="#contentToHide" checked>
        <label class="form-check-label" for="hideCheckbox">
            Hide things
        </label>
    </div>

    <div class="collapse mt-3" id="contentToHide">
        <p>Dies ist der Inhalt, der versteckt werden soll.</p>
    </div>
</div>

<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"></script>

</body>
</html>
```

### PR DESCRIPTION
**Pull request by TommasoAllegretti**
### Description

The Collapse documentation only talks about buttons and <a> elements as toggles. However, using checkboxes also works fine. Therefore, it should be pointed out in the docs that you can also use checkboxes as collapse toggles.

### Motivation & Context

As requested in issue #39728 

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

#### Live previews

- <https://deploy-preview-39767--twbs-bootstrap.netlify.app/>

### Related issues

Closes #39728

### CODE REVIEW

**Comment by julien-deramond** [comment]
I'd say that we shouldn't mention checkboxes specifically but rather make it understandable that collapse can be used with basically everything. For example, you can also use it with basic HTML elements such as `<code>`, `<div>`, etc.

```html
<code data-bs-toggle="collapse" data-bs-target="#collapseExample">Test from code</code>
<div data-bs-toggle="collapse" data-bs-target="#collapseExample" class="w-100 h-100 bg-warning">Test from div</div>
<div class="collapse" id="collapseExample" style="">
  <div class="card card-body">
    Some placeholder content for the collapse component. This panel is hidden by default but revealed when the user activates the relevant trigger.
  </div>
</div>
```

Maybe we can make it clearer in "How it works" that it works with everything and that buttons and links are used on this page because they would be the most common elements as triggers 🤷

**Comment by XhmikosR** [comment]
Agreed, we are better off documenting the above. Let alone that the checkbox in this PR is hard to notice next to the buttons.

**Comment by TommasoAllegretti** [comment]
Sorry for the delayed response, I followed your feedback and changed the documentation accordingly.
Please let me know if the documentation is now clear enough in explaining the possibility of using various different elements to toggle a collapse.

**Comment by patrickhlauke** [comment]
as an aside, we probably don't want to suggest using an arbitrary non-focusable element, for accessibility reasons

File: site/content/docs/5.3/components/collapse.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ toc: true
 
 ## How it works
 
-The collapse JavaScript plugin is used to show and hide content. Buttons or anchors are used as triggers that are mapped to specific elements you toggle. Collapsing an element will animate the `height` from its current value to `0`. Given how CSS handles animations, you cannot use `padding` on a `.collapse` element. Instead, use the class as an independent wrapping element.
+The collapse JavaScript plugin is used to show and hide content. Buttons, anchors and many other elements can be used as triggers that are mapped to specific elements you toggle. Collapsing an element will animate the `height` from its current value to `0`. Given how CSS handles animations, you cannot use `padding` on a `.collapse` element. Instead, use the class as an independent wrapping element.
```
Comment (patrickhlauke) [inline site/content/docs/5.3/components/collapse.md:11]: other *interactive*/*focusable* elements (while technically that's not required in the code, it'd be dangerous to suggest authors can just make anything a trigger, for accessibility reasons.

File: site/content/docs/5.3/components/collapse.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Click the buttons below to show and hide another element via class changes:
 - `.collapsing` is applied during transitions
 - `.collapse.show` shows content
 
-Generally, we recommend using a `<button>` with the `data-bs-target` attribute. While not recommended from a semantic point of view, you can also use an `<a>` link with the `href` attribute (and a `role="button"`). In both cases, the `data-bs-toggle="collapse"` is required.
+Generally, we recommend using a `<button>` with the `data-bs-target` attribute. While not recommended from a semantic point of view, you can also use an `<a>` link with the `href` attribute (and a `role="button"`), an input with `type="checkbox"` or even a `<div>` tag. In all cases, the `data-bs-toggle="collapse"` is required.
```
Comment (patrickhlauke) [inline site/content/docs/5.3/components/collapse.md:25]: again, just having a `<div>` will make it an inaccessible control, so probably best not to suggest that.

**Review by patrickhlauke — CHANGES_REQUESTED** [review]

**Comment by mdo** [comment]
Agreed on the advice of only encouraging semantic elements for this kind of thing. Sure a checkbox can be used, and any other element, but I don't think that needs stating here. We encourage one thing—we don't need to encourage everything IMO. Anyone on the maintainer team though that feels strongly is free to overrule though :).

**Comment by carlosmintfan** [comment]
Are you sure? A checkbox looks much better design for deciding whether some elements in a list should be shown or not.

**Pull request closed without merging**